docs/core/extensions/logging.md

```diff
@@ -169,6 +241,34 @@ On [Azure App Service](https://azure.microsoft.com/services/app-service/), selec
 
 For more information on setting .NET configuration values using environment variables, see [environment variables](configuration-providers.md#environment-variable-configuration-provider).
 
+### Configure logging with code
+
+Logging is configured using the <xref:Microsoft.Extensions.Logging.ILoggingBuilder> API. This can be accessed from different places:
+
+- If creating the `ILoggerFactory` directly, <xref:Microsoft.Extensions.Logging.LoggerFactory.Create%2A?displayProperty=nameWithType>
+- If using DI without a host, <xref:Microsoft.Extensions.DependencyInjection.LoggingServiceCollectionExtensions.AddLogging%2A?displayProperty=nameWithType>
+- If using a host, <xref:Microsoft.Extensions.Hosting.HostApplicationBuilder.Logging?displayProperty=nameWithType>, <xref:Microsoft.AspNetCore.Builder.WebApplicationBuilder.Logging?displayProperty=nameWithType> or other host specific APIs
+
+This example shows setting the console [logging provider](logging-providers.md) and several [filters](#how-filtering-rules-are-applied).
+
+```csharp
+using var loggerFactory = LoggerFactory.Create(builder =>
+{
+    builder
+        .AddFilter("Microsoft", LogLevel.Warning)
+        .AddFilter("System", LogLevel.Warning)
+        .AddFilter("LoggingConsoleApp.Program", LogLevel.Debug)
+        .AddConsole();
+});
+
+ILogger logger = loggerFactory.CreateLogger<Program>();
+logger.LogDebug("Example log message");
```

Comment:
Minor, can we change this to a message template with one or two key-value pairs? (to avoid readers interpreting it as logging a raw string)